ui/lib/config-ui/addon/components/messages/page/list.hbs

```diff
@@ -0,0 +1,113 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<Messages::TabPageHeader @authenticated={{@authenticated}}>
+  <:toolbarActions>
+    <ToolbarLink data-test-toolbar-create-message @route="messages.create" @type="add">
+      Create message
+    </ToolbarLink>
+  </:toolbarActions>
+</Messages::TabPageHeader>
+
+{{#if @messages.length}}
+  {{#each @messages as |message|}}
+    <LinkedBlock
+      data-test-list-item={{message.id}}
+      class="list-item-row"
+      @params={{array "messages.message.details" message.id}}
+      @linkPrefix={{this.mountPoint}}
+    >
+      <div class="level is-mobile">
+        <div class="level-left">
+          <div>
+            <Hds::Text::Display @tag="h2" data-linked-block-title={{message.id}}>
+              <Icon @name="message-circle" class="auto-width" aria-label="message" />
+              {{message.title}}
+            </Hds::Text::Display>
+            <div class="has-top-margin-xs">
+              <Hds::Badge
+                @text={{if
```

Comment:
I'm a little hesitant about doing all the string logic for showing certain texts based on date/time and message active status in the template, but noticed that our current pattern is to do it this way. My other thought would be to do it in the model (but this can get messy).

Reply:
Good thought. My process is always what is the easiest to read for a future developer. Looking at the logic, it looks like this might be a good candidate for the component js file because it's hard to read. And I would only put it on the model if this was going to be used in other places.

Reply:
This is hard to be done in the current component file since it's an individual message. I would need to create another component that passes the `message` in order to do this logic in there. Not sure if this stuff in the LinkedBlock needs its own component 🤔